server/app/services/program/ProgramService.java

```diff
@@ -814,12 +840,28 @@ public ErrorAnd<ProgramDefinition, CiviFormError> appendStatus(
                 .add(status)
                 .build());
 
-    return ErrorAnd.of(
-        syncProgramDefinitionQuestions(
-                programRepository.getShallowProgramDefinition(
-                    programRepository.updateProgramSync(program.toProgram())))
-            .toCompletableFuture()
-            .join());
+    ErrorAnd<ProgramDefinition, CiviFormError> result =
+        ErrorAnd.of(
+            syncProgramDefinitionQuestions(
+                    programRepository.getShallowProgramDefinition(
+                        programRepository.updateProgramSync(program.toProgram())))
+                .toCompletableFuture()
+                .join());
+    if (!result.isError()) {
+      // Temporary hook to create a new row in ApplicationStatuses table for the status change.
+      // Remove these hooks when we have created service class for creating/updating
+      // statuses from program controllers.
+      // github issue- https://github.com/civiform/civiform/issues/7040
+      applicationStatusesRepository.createOrUpdateStatusDefinitions(
+          program.adminName(),
+          new StatusDefinitions()
+              .setStatuses(
+                  ImmutableList.<StatusDefinitions.Status>builder()
```

Comment:
**suggestion (non-blocking):** You can consider pulling this out into a variable and using it here and above, but since this is temporary, feel free to leave it as is if you prefer.

Reply:
It get moved Danielle soon. I think its ok to leave it here. :)